Advancing_front_surface_reconstruction/examples/Advancing_front_surface_reconstruction/reconstruction_structured.cpp

```diff
@@ -0,0 +1,185 @@
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/Point_with_normal_3.h>
+#include <CGAL/Shape_detection_3.h>
+#include <CGAL/structure_point_set.h>
+#include <CGAL/Delaunay_triangulation_3.h>
+#include <CGAL/Triangulation_vertex_base_with_info_3.h>
+#include <CGAL/Advancing_front_surface_reconstruction.h>
+#include <CGAL/IO/read_xyz_points.h>
+
+#include <boost/lexical_cast.hpp>
+
+typedef CGAL::Exact_predicates_inexact_constructions_kernel  Kernel;
+typedef Kernel::Point_3  Point;
+typedef std::pair<Kernel::Point_3, Kernel::Vector_3>         Point_with_normal;
+typedef std::vector<Point_with_normal>                       Pwn_vector;
+typedef CGAL::First_of_pair_property_map<Point_with_normal>  Point_map;
+typedef CGAL::Second_of_pair_property_map<Point_with_normal> Normal_map;
+
+// Efficient RANSAC types
+typedef CGAL::Shape_detection_3::Efficient_RANSAC_traits
+  <Kernel, Pwn_vector, Point_map, Normal_map>                Traits;
+typedef CGAL::Shape_detection_3::Efficient_RANSAC<Traits>    Efficient_ransac;
+typedef CGAL::Shape_detection_3::Plane<Traits>               Plane;
+
+// Point set structuring type
+typedef CGAL::Point_set_with_structure<Traits>               Structure;
+
+// Advancing front types
+typedef CGAL::Advancing_front_surface_reconstruction_vertex_base_3<Kernel> LVb;
+typedef CGAL::Advancing_front_surface_reconstruction_cell_base_3<Kernel> LCb;
+typedef CGAL::Triangulation_data_structure_3<LVb,LCb> Tds;
+typedef CGAL::Delaunay_triangulation_3<Kernel,Tds> Triangulation_3;
+typedef Triangulation_3::Vertex_handle Vertex_handle;
+
+typedef CGAL::cpp11::array<std::size_t,3> Facet;
+
+
+// Functor to init the advancing front algorithm with indexed points
+struct On_the_fly_pair{
+  const Pwn_vector& points;
+  typedef std::pair<Point, std::size_t> result_type;
+
+  On_the_fly_pair(const Pwn_vector& points) : points(points) {}
+  
+  result_type
+  operator()(std::size_t i) const
+  {
+    return result_type(points[i].first,i);
+  }
+};
+
+// Specialized priority functor that favor structure coherence
+template <typename Structure>
+struct Priority_with_structure_coherence {
```

Comment:
It is just one way of doing it and it is mostly empirical (not publication and no scientific proof that it works). I thought that was a bit "superficial" to be a proper CGAL feature but that it was still a nice thing to show as an example.